### PR TITLE
Feature/unittest

### PR DESCRIPTION
**Pull request by alen3003**

### CODE REVIEW

File: WeatherPlannerDemoTests/Features/CityDetails/CityDetailsPresenterTests.swift [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    private var wind: City.WindInfo!
+    private var coordination: City.Coordination!
+    private var city: City!
+    private var cityViewModel: CityViewModel!
+
+    override func setUpWithError() throws {
+        Resolver.custom.register { CityDetailsUseCaseMock() }.implements(CityDetailsUseCaseProtocol.self)
+        Resolver.custom.register { CoreDataStackMock() }.implements(CoreDataStackProtocol.self)
+        
+        temperature = City.TemperatureInfo(
```
Comment (labasbozidar) [inline WeatherPlannerDemoTests/Features/CityDetails/CityDetailsPresenterTests.swift]: You can create these models in separate file with static properties.

File: WeatherPlannerDemoTests/Features/CityList/CityListPresenterTests.swift [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    let disposeBag = DisposeBag()
+
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+        
+        Resolver.custom.register { CoordinatorMock() }.implements(CoordinatorProtocol.self)
+        Resolver.custom.register { CityListUseCaseMock() }.implements(CityListUseCaseProtocol.self)
+        Resolver.custom.register { CoreDataStackMock() }.implements(CoreDataStackProtocol.self)
+        
+        presenter = CityListPresenter()
```
Comment (labasbozidar) [inline WeatherPlannerDemoTests/Features/CityList/CityListPresenterTests.swift]: Can you use something like:
```
presenter = Resolver.custom.resolve()
```

File: WeatherPlannerDemoTests/Sourcery/Generated/AutoMockable.generated.swift [hunk 3]
```diff
@@ -0,0 +1,61 @@
+// Generated using Sourcery 1.0.2 — https://github.com/krzysztofzablocki/Sourcery
+// DO NOT EDIT
+
+// swiftlint:disable variable_name
+// swiftlint:disable line_length
+
+import RxSwift
+@testable import WeatherPlannerDemo
+class CityDetailsUseCaseProtocolMock: CityDetailsUseCaseProtocol {
```
Comment (labasbozidar) [inline WeatherPlannerDemoTests/Sourcery/Generated/AutoMockable.generated.swift:9]: Where did you use these mock classes?

**Review by labasbozidar — CHANGES_REQUESTED** [review]

File: WeatherPlannerDemoTests/Sourcery/Generated/AutoMockable.generated.swift (hunk 3 above)
Comment (alen3003) [reply WeatherPlannerDemoTests/Sourcery/Generated/AutoMockable.generated.swift:9]: I have a really unique case where I have to create CoreData objects with context, so auto mocked classes do not work for me, however in terms of practice I have left them in project, until I figure out a better use for them

**Review by labasbozidar — APPROVED** [review]

**Pull request merged**